test/pkg/compose/templatingdata/assembler.go

```diff
@@ -0,0 +1,392 @@
+package templatingdata
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/fs"
+	"net/http"
+	"os"
+	"path"
+	"path/filepath"
+	"regexp"
+	"runtime"
+	"strconv"
+	"strings"
+
+	"github.com/openshift/microshift/pkg/util"
+	"github.com/openshift/microshift/test/pkg/testutil"
+	"k8s.io/klog/v2"
+)
+
+var (
+	releaseInfoRpmRx   = regexp.MustCompile("^microshift-release-info-.*.rpm$")
+	errNoRemoteRelease = fmt.Errorf("release from remote repo not found")
+)
+
+type TemplatingDataOpts struct {
+	ArtifactsMainDir               string
+	TemplatingDataFragmentFilepath string
+	SkipContainerImagesExtraction  bool
+}
+
+func (o *TemplatingDataOpts) Construct() (*TemplatingData, error) {
+	klog.InfoS("Constructing TemplatingData")
+
+	rpmRepos := path.Join(o.ArtifactsMainDir, "rpm-repos")
+	localRepo := path.Join(rpmRepos, "microshift-local")
+	fakeNextRepo := path.Join(rpmRepos, "microshift-fake-next-minor")
+	baseRepo := path.Join(rpmRepos, "microshift-base")
+	externalRepo := path.Join(rpmRepos, "microshift-external")
```

Comment:
This feels like another opportunity to be more data driven, instead of building the list of repositories into the app. Maybe a config file? Or a set of files, like the package source files? Then the TemplatingData struct would need a map from from the repo name to the path and any other details that go into the template.

Reply:
That's possible, but it would cause the templating data to be more dynamic (map[string]interface{}) since we won't be able to access the data in the templates by struct members names - like `.Current.Repository`, instead if would be `.data[X].Repository`

Reply:
That's reasonable.